src/ciadmin/check/check_privileged.py

```diff
@@ -36,6 +38,35 @@ async def check_privileged_is_untrusted(generate_resources):
                 .get("allowPrivileged", False)
             ):
                 privileged = True
+                break
+            if gwConfig := launchConfig.get("workerConfig", {}).get(
+                "genericWorker", {}
+            ):
+                # generic-worker
+                if gwConfig.get("config", {}).get("enableRunTaskAsCurrentUser", True):
+                    privileged = True
+                    break
+                if (
+                    gwConfig.get("config", {})
+                    .get("d2gConfig", {})
+                    .get("enableD2G", False)
+                ):
+                    d2gConfig = gwConfig["config"]["d2gConfig"]
+                    if d2gConfig.get("allowPrivileged", True):
+                        privileged = True
+                        break
+                    if d2gConfig.get("allowGPUs", True):
```

Comment:
`allowGPUs` defaults to `False`

Should be:


```suggestion
                    if d2gConfig.get("allowGPUs", False):
```

Reply:
Also, why is this considered privileged, just wondering?

Reply:
While the default is currently False in g-w, I think it's safer / more future-proof to explicitly disable it here for trusted pools.

As to why I'd consider it privileged, gpus add a massive attack surface so at this stage I'd prefer to blanket disallow them for trusted pools at least until such time as there's a use case and we need to revisit.